TypeCobol/Compiler/Scanner/Scanner.cs

```diff
@@ -959,11 +959,15 @@ public Token GetNextToken()
                 case ':':
                     // -- TypeCobol specific syntax --
                     // QualifiedNameSeparator => qualifierName::qualifiedName
-                    if (currentIndex < lastIndex && line[currentIndex + 1] == ':')
+                    if (currentIndex < lastIndex && line[currentIndex + 1] == ':' && (this.compilerOptions == null || !this.compilerOptions.AreForCopyParsing))
                     {
```

Comment:
Can you try the following solution:
Modify this condition:
```C#
if (!currentState.InsidePseudoText //No QualifiedNameSeparator allowed in pseudoText
                        && (we are not in a COPY) //No QualifiedNameSeparator allowed in COPY 
                        && currentIndex< lastIndex && line[currentIndex + 1] == ':')
                    {
```

Delete line 2307 in original Scanner.cs, because now we allow empty pseudoText `==::==`
```C#
            // no legal cobol word chars found 
            if (index == startIndex + 1 && !CobolChar.IsCobolWordChar(line[index])) return false;
```
Keep your fix in TypeCobol/Compiler/Scanner/Token.cs
Still return a bool from method BuildReplaceOperation in TypeCobol/Compiler/CupPreprocessor/CompilerDirectiveBuilder.cs to report Diagnostic : "REPLACE Empty Comparison Pseudo Text." and "REPLACE Empty Pseudo Text Delimiter"

Otherwise all of the rest of the C# code can be deleted.

I didn't test Codegen, but it worked on Parser part.

Reply:
Yes it works like that!

Reply:
I am going to create a new PR and give up this one. I didn't know Partial Cobol Word pattern and how it works :-|